### ISSUE TITLE
dataset_testA_for_track23.json

### ISSUE THREAD

**Issue by 277CPS** [WENGSYX/CMIVQA_Baseline#11]
这个是要按照get_data中的函数自己生成吗

**Comment by 277CPS** [comment]
以及main是跑通了的，设备问题应该能满足，test中检查了很多断点，数据也应该是读进去了。出现了以上的问题，test没有给出参数，想请问一下test的参数，以及您使用的设备条件。

**Comment by 277CPS** [comment]
a, b = torch.stack(logits).sort(descending=True)
test.py总在这里被killed

**Comment by WENGSYX** [comment]
Hi,

这个并不是自己生成的，而是作为官方的测试文件。此文件系NLPCC2023任务中所提供的TestA数据集文件，而NLPCC 2024任务中我们将在整体评测时期提供新的多语言数据集文件，敬请期待。

**Comment by WENGSYX** [comment]
关于a, b = torch.stack(logits).sort(descending=True)

你的代码被killed可能是由于内存不足的问题，尤其是在排序大量 logits 时。

### 批量处理
将 logits 的处理分成较小的批次。这样可以减少一次性处理的数据量，从而降低内存消耗。

```
BATCH_SIZE = 10  # 根据内存情况调整这个值

def test_model(model, val_loader):
    model.eval()
    video_logits = []
    outputs = []
    tk = tqdm(val_loader, total=len(val_loader), position=0, leave=True)
    for step, (input_ids, attention_mask, token_types, vfeats, vfeats_mask) in enumerate(tk):
        logits = []
        ns_dict = {}
        ns = 0
        with torch.no_grad():
            for i in range(len(input_ids)):
                output = model.forward_test(
                    input_ids=input_ids[i].unsqueeze(dim=0),
                    attention_mask=attention_mask[i].unsqueeze(dim=0),
                    token_types=token_types[i].unsqueeze(dim=0),
                    vfeats=vfeats[i].unsqueeze(dim=0),
                    vfeats_mask=vfeats_mask[i].unsqueeze(dim=0)
                )
                ls = output['logits'].view(-1)
                for ln in range(len(ls)):
                    ns_dict[ns] = [i, int(ln / output['logits'].size(0)), int(ln % output['logits'].size(0))]
                    ns += 1
                logits.extend(ls)
                video_logits.append(output['logits'][output['start'], output['end']])

            logits = torch.stack(logits)
            # 分批排序
            batch_sorted_indices = []
            for start in range(0, len(logits), BATCH_SIZE):
                end = start + BATCH_SIZE
                batch_logits = logits[start:end]
                batch_sorted_indices.extend(torch.sort(batch_logits, descending=True)[1].tolist())
            b = torch.tensor(batch_sorted_indices)
            ops = []
            for n in b[:50]:
                ops.append([video_ids[ns_dict[n.item()][0]]] + get_time(ns_dict[n.item()][1], ns_dict[n.item()][2], input_ids[ns_dict[n.item()][0]], video_ids[ns_dict[n.item()][0]]))
            outputs.append({'id': step, 'question': valid[step]['question'], 'output': ops})

    return outputs

```
